src/util/sort_key.rs

```diff
@@ -0,0 +1,85 @@
+// Rust Bitcoin Library
+// Written in 2020 by
+//     rust-bitcoin developers
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Utility for key Sorting
+//!
+//! Sort `bitcoin::PublicKey` according to the logic implemented in
+//! bitcoin core. In bitcoin core, keys are merely wrappers over
+//! bytes and the comparison function directly checks the underlying
+//! bytes.
+//! When keys are compressed, this utility can be used for BIP67
+//! comparison.
+use std::cmp;
+
+use PublicKey;
+
+/// Utility structure for representing ordering of PublicKeys
+/// as implemented in bitcoin core. Compares the underlying
+/// bytes when comparing the two PublicKeys in compressed form
+/// instead of naturally derived order which compares underlying
+/// 64 byte secp key.
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct SortKey<'pk>(&'pk PublicKey);
```

Comment:
In fact to make this a bit more ergonomic, this could be a `Cow<PublicKey>` so that this can also be used in owned form (like to build a map structure without having to keep the keys separately).